test-clients/src/main/java/com/hedera/services/bdd/spec/verification/traceability/SidecarWatcher.java

```diff
@@ -82,15 +91,17 @@ public void watch() throws IOException {
                 final var ev = (WatchEvent<Path>) event;
                 final var filename = ev.context();
                 final var child = recordStreamFolderPath.resolve(filename);
-                log.debug("Record stream file created -> {} ", child.getFileName());
-                final var newFilePath = child.toAbsolutePath().toString();
+                final var newFilePath = child.toString();
                 if (SIDECAR_FILE_REGEX.matcher(newFilePath).find()) {
-                    log.info("We have a new sidecar.");
-                    final var sidecarFile = RecordStreamingUtils.readSidecarFile(newFilePath);
-                    onNewSidecarFile(sidecarFile);
-                    if (shouldTerminateAfterNextSidecar) {
-                        return;
+                    if (firstSidecarNotifyEventSeen) {
```

Comment:
Q: If each side car gets 2 notifications is this parsing getting run twice for every file after the 1st?

Reply:
[We toggle the flag after each sidecar notification](https://github.com/hashgraph/hedera-services/pull/3890/files#diff-0656ea47a120bd42fc6adf0fac25b0d126ef80bd704e78a752988a5d0a1cf4bcR104), independent of whether we parsed the file or not (i.e. on both first and second events). 

So, on the first event for each sidecar file, we don't go into the if and we do not parse the file (since the flag is `false`), but we still toggle the flag to `true`. Then, on the second event, we go into the if (since the flag is `true`), do the parsing and all other stuff, and then we again toggle the flag at the end, setting it to `false`. And this whole process repeats 

I agree the name of the flag was confusing... changed it now to `firstEventOfSidecarReceived`, not to confuse the reader that it is tracking events for only the first sidecar file.